From 730df16e51923cb373a9796dabbd4ce4d2c447e8 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Fri, 26 Jul 2013 12:23:54 -0400
Subject: [PATCH 51/99] Ticket 47427 - limits check reported wrong value

Description:  There was a formating error that displayed the wrong value for LONG_MAX

https://fedorahosted.org/389/ticket/47427
(cherry picked from commit e456f868f91d1075777a2ac43fb8145fdd289036)
---
 ldap/servers/slapd/libglobs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ldap/servers/slapd/libglobs.c b/ldap/servers/slapd/libglobs.c
index cf1f73e..413351d 100644
--- a/ldap/servers/slapd/libglobs.c
+++ b/ldap/servers/slapd/libglobs.c
@@ -1239,8 +1239,8 @@ config_set_disk_threshold( const char *attrname, char *value, char *errorbuf, in
     threshold = strtoll(value, &endp, 10);
 
     if ( *endp != '\0' || threshold < 4096 || errno == ERANGE ) {
-        PR_snprintf ( errorbuf, SLAPI_DSE_RETURNTEXT_SIZE, "%s: \"%s\" is invalid, threshold must be greater than 4096 and less then %lu",
-            attrname, value, LONG_MAX );
+        PR_snprintf ( errorbuf, SLAPI_DSE_RETURNTEXT_SIZE, "%s: \"%s\" is invalid, threshold must be greater than 4096 and less then %lld",
+            attrname, value, (long long int)LONG_MAX );
         retVal = LDAP_OPERATIONS_ERROR;
         return retVal;
     }
-- 
1.8.1.4

